### ISSUE TITLE
[Clarification] Last IMU device as root for normalization?

### ISSUE THREAD

**Issue by bryanbocao** [ZuoCX1996/Loose-Inertial-Poser#11]
Dear Authors,

Thanks for sharing the code again. It's been helpful in my research.

I am trying to understand the **rotation to the root coordinate system** part in ```LipData class``` in

https://github.com/ZuoCX1996/Loose-Inertial-Poser/blob/10b480bc2c55499be5849cc24b7a19f4f4e81e8b/my_data.py#L456-L464

I prepared the shapes to help describing my question:
```
        print(f'\npp460 acc.size(): {acc.size()}') # pp460 acc.size(): torch.Size([424998, 4, 3])
        acc = torch.cat((acc[:, :3] - acc[:, 3:], acc[:, 3:]), dim=1).bmm(rot[:, -1]) / 30
        # 转换为相对根节点的旋转
        print(f'\npp463 rot.size(): {rot.size()}') # pp463 rot.size(): torch.Size([424998, 4, 3, 3])
        rot = torch.cat((rot[:, 3:].transpose(2, 3).matmul(rot[:, :3]), rot[:, 3:]), dim=1)
        print(f'\npp465 rot.size(): {rot.size()}') # pp465 rot.size(): torch.Size([424998, 4, 3, 3])

        # 归一化
        print(f'\npp469 joint.size(): {joint.size()}') # pp469 joint.size(): torch.Size([424998, 24, 3])
        joint = joint - joint[:, :1, :]
        print(f'\npp474 joint.size(): {joint.size()}') # pp474 joint.size(): torch.Size([424998, 24, 3])

        # 转到root坐标系
        joint = joint.bmm(rot[:, -1])
        print(f'\npp477 joint.size(): {joint.size()}') # pp477 joint.size(): torch.Size([424998, 24, 3])
```

**Q1**: The code assumes that the last one of the ```4``` IMU devices serves as the root for ```acc``` and ```rot```:
```
acc = torch.cat((acc[:, :3] - acc[:, 3:], acc[:, 3:]), dim=1).bmm(rot[:, -1]) / 30
rot = torch.cat((rot[:, 3:].transpose(2, 3).matmul(rot[:, :3]), rot[:, 3:]), dim=1)
```
**Q2**: Similarly, the following code treats the last joint as root:
```
        joint = joint - joint[:, :1, :]
        joint = joint.bmm(rot[:, -1])
```

Is this understanding correct? Thank you for taking your time in advance.

**Comment by ZuoCX1996** [comment]
Hi,

Your understanding is correct!

For Q1: We indeed use the last of the 4 IMU devices as the root node for both acceleration (acc) and rotation (rot) processing. This last IMU typically corresponds to the waist/pelvis region, which serves as a stable reference point for the coordinate system.

For Q2: Similarly, the processing of joint coordinates treats the last joint as the root node. The operations joint = joint - joint[:, :1, :] and joint = joint.bmm(rot[:, -1]) are designed to normalize the joint positions relative to this root and transform them into the root coordinate system.

**Comment by bryanbocao** [comment]
Thanks for the clarification!
